src/plugins/intel_npu/src/compiler/include/zero_compiler_in_driver.hpp

```diff
@@ -177,30 +185,15 @@ class LevelZeroCompilerInDriver final : public ICompiler {
 };
 
 template <typename TableExtension>
-LevelZeroCompilerInDriver<TableExtension>::LevelZeroCompilerInDriver(const char* extName,
-                                                                     ze_driver_handle_t driverHandle)
+LevelZeroCompilerInDriver<TableExtension>::LevelZeroCompilerInDriver(ze_driver_handle_t driverHandle,
+                                                                     ze_device_handle_t deviceHandle,
+                                                                     ze_context_handle_t zeContext,
+                                                                     ze_graph_dditable_ext_last_t* graph_ddi_table_ext)
     : _driverHandle(driverHandle),
+      _graphDdiTableExt(reinterpret_cast<TableExtension*>(graph_ddi_table_ext)),
       _logger("LevelZeroCompilerInDriver", Logger::global().level()) {
-    // Load our graph extension
-    auto result =
-        zeDriverGetExtensionFunctionAddress(_driverHandle, extName, reinterpret_cast<void**>(&_graphDdiTableExt));
-
-    if (ZE_RESULT_SUCCESS != result) {
-        OPENVINO_THROW("Failed to initialize zeDriver. Error code: ", std::hex, result);
-    }
-
-    uint32_t deviceCount = 1;
-    // Get our target device
-    result = zeDeviceGet(_driverHandle, &deviceCount, &_deviceHandle);
-    if (ZE_RESULT_SUCCESS != result) {
-        OPENVINO_THROW("Failed to get device. Error code: ", std::hex, result);
-    }
-
-    ze_context_desc_t contextDesc = {ZE_STRUCTURE_TYPE_CONTEXT_DESC, nullptr, 0};
-    result = zeContextCreate(_driverHandle, &contextDesc, &_context);
-    if (ZE_RESULT_SUCCESS != result) {
-        OPENVINO_THROW("Failed to initialize context for device. Error code: ", std::hex, result);
-    }
+    _context = zeContext;
+    _deviceHandle = deviceHandle;
```

Comment:
Move `_context` and `_deviceHandle `above. Maybe it can be done in the ctor declaration as well.

Reply:
yes, update this in the latest commit

Reply:
Is there a specific reason for keeping this implementation in the .hpp file? Can we just move it to .cpp file or remove the logger and move it to the class definition?

Reply:
ya, i think we can move this implementation into the `zero_compiler_in_driver.cpp`,
updated in the latest commit, thanks